bin/udb3.php

```diff
@@ -95,6 +97,14 @@
 $consoleApp->add(new ValidatePlaceJsonLdCommand($app['event_command_bus']));
 $consoleApp->add(new MarkPlaceAsDuplicateCommand($app['event_command_bus'], $app[LocationMarkedAsDuplicateProcessManager::class]));
 $consoleApp->add(new DispatchMarkedAsDuplicateEventCommand($app['event_command_bus'], $app[LocationMarkedAsDuplicateProcessManager::class], $app['event_bus']));
+$consoleApp->add(
+    new ReindexOffersWithPopularityScore(
+        $app['dbal_connection'],
+        $app['event_bus'],
+        $app[EventJSONLDServiceProvider::JSONLD_PROJECTED_EVENT_FACTORY],
+        $app[PlaceJSONLDServiceProvider::JSONLD_PROJECTED_EVENT_FACTORY]
+    )
+);
```

Comment:
Maybe we could instantiate the same `ReindexOffersWithPopularityScore` command class twice, once for `events` and once for `places`, and instead of passing in two factories we only pass the relevant one + a string for the offer type?

So we'd end up with two commands like `place:reindex-places-with-popularity` and `event:reindex-events-with-popularity`, instead of one `offer:reindex-offers-with-popularity` with two factories and a required type factory, forcing us to do some if/else checks in the command class.

Or alternatively, we could keep it like it is but make the `type` option optional and determine which factory to use for which result by looking at the type in the database row. Then we still have the two factories and if/else checks, but it would be more flexible.

In any case this is just a suggestion for a slight improvement that popped into my head while reviewing, not a blocker. It also works as it is right now and can always be refactored later if we do need to split it up or make it more flexible for some reason.

Reply:
With the first proposal we would have 2 CLI commands which makes it possible to re-index either places or events.
With the second proposal I don't think we can decide which offer type to re-index?

I think it makes sense to re-index either places or events. Because a re-index of places could fix some events.

Reply:
I found the first proposal combining a flexible behaviour with an elegant implementation, so I did refactor it to that proposal. I also used the `OfferType` enum.

Reply:
Either way is fine by me, I just want to clarify something about:

> Because a re-index of places could fix some events.

Possibly, but if you run `event:reindex-events-with-popularity` afterwards it will again loop over all events in the popularity table regardless if they have already been indexed as a side effect of re-indexing the places no?

Reply:
Also:

> With the second proposal I don't think we can decide which offer type to re-index?

The second proposal was to make the `type` option _optional_, so we could choose to either run it for both events + places (don't pass the `--type`) or for one or the other (`--type=event` or `--type=place`)

Reply:
But anyway your implementation achieves the intended goal so it's all nitpicking. 😄 They were just some suggestions that could either make the command do less if/else checks based on the type by injecting the type (as string, for configuring the name) + the right factory (suggestion 1), or that would make the command more flexible (by allowing the user to choose to run it for either all offers or a specific type). The other comments were more important and those are resolved so this looks good to me! The rest can always be refactored if ever needed.

Reply:
Ah I see you even actually implemented suggestion no. 2! Thanks, looks great 👌